### PR TITLE
Pr for ticket #123315 

### PR DESCRIPTION
**Pull request by dleyba042**
This PR makes changes to the _actionHandler function used on the Slider.Dart Widget for Key Events. It ensures onChangeStart is called at the beginning of a Key Event and onChangeEnd at the end of one. This PR includes a test for the changes made.
I ran all existing tests after my changes were made and they passed.

### CODE REVIEW

**Comment by dwgray** [comment]
Devin - this looks really good.  I am going to spend a few minutes adding some comments. But aside from the question about how dart switch statements work, I wouldn't consider any of them blocking.  As I've noted elsewhere, I'd like everyone to pile one with comments and questions - code reviews are a great way to learn.

File: packages/flutter/lib/src/material/slider.dart [hunk 1]
```diff
@@ -684,26 +684,41 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
   void _actionHandler(_AdjustSliderIntent intent) {
     final _RenderSlider renderSlider = _renderObjectKey.currentContext!.findRenderObject()! as _RenderSlider;
     final TextDirection textDirection = Directionality.of(_renderObjectKey.currentContext!);
+    bool sliderIncreased = false;
+    widget.onChangeStart?.call(widget.value);
+
     switch (intent.type) {
```
Comment (dwgray) [inline packages/flutter/lib/src/material/slider.dart:688]: Question:  When a case statement doesn't end with a break, most C-style languages either fall through or error at compile time.  From my reading of this https://dart.dev/language/control-flow#switch-and-case - I'd expect a compiler error here.  And if it falls through, I'd expect some really weird behavior.  Even through you didn't introduce this, I'd like to understand what is going on here as a Dart newbie.

File: packages/flutter/lib/src/material/slider.dart [hunk 2]
```diff
@@ -684,26 +684,41 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
...
           case TextDirection.ltr:
             renderSlider.decreaseAction();
         }
       case _SliderAdjustmentType.up:
         renderSlider.increaseAction();
+        sliderIncreased = true;
       case _SliderAdjustmentType.down:
         renderSlider.decreaseAction();
     }
+
```
Comment (dwgray) [inline packages/flutter/lib/src/material/slider.dart]: This is almost completely a matter of style, but I'd consider playing with this if statement and seeing if you could make it cleaner.  I don't like the fact that line 716 and line 720 are almost completely identical (it's hard to parse out the difference).   Some thought would be to lose the if statement and use a ternary operator to distinguish between the increased/decreased case inline in the parameter to campDouble.  Or to pull out the computation of that value to a separate statement and do something like
```
var unit = renderSlider._semanticActionUnit,0.0,1.0);
var change = renderSlider.value + (sliderIncreased ? : unit : - unit);
widget.onChangeEnd?.call(_lerp(clampDouble(change)));
```

File: packages/flutter/test/material/slider_test.dart [hunk 3]
```diff
@@ -2375,6 +2375,99 @@ void main() {
     expect(bottomSliderValue, 0.5, reason: 'unfocused bottom Slider unaffected by third arrowRight');
   });
 
+testWidgets('When adjusting slider with arrow keys onChangeStart and onChangeEnd callbacks are activated and display correct values', (WidgetTester tester) async
```
Comment (dwgray) [inline packages/flutter/test/material/slider_test.dart]: You actually fixed several things in this change and are testing them all in one test.  It's generally best practice in unit testing to test one thing per test - then when the test fails you can more easily go right to the problem.  I'd consider taking each of the places where you have a cluster of asserts and make that a separate test (possibly abstracting out a helper function to reduce redundant code for setting up the tests).

This is a very light suggestion.  It looks like the Flutter team doesn't do a great job of abiding by the principal I noted.  They may prefer to make sure they have coverage, so at a quick glance I'd guess this meets Flutter's standards.

File: packages/flutter/lib/src/material/slider.dart [hunk 4]
```diff
@@ -684,26 +684,41 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
...
+    if(sliderIncreased)
+    {
+      widget.onChangeEnd?.call(_lerp(clampDouble(renderSlider.value  + renderSlider._semanticActionUnit,0.0,1.0)));
+    }
```
Comment (dwgray) [inline packages/flutter/lib/src/material/slider.dart]: Question:  Do you know at what point in the process this code goes through style checking?  I notice two spaces in front of the + on line 716, which I'd expect something to catch and/or correct.

**Review by dwgray — COMMENTED** [review]
As I noted before, these comments are all basically questions.  I'd really like to know what is going on with Dart switch/case statements but the rest are basically stylistic.  I've never used github for review before, so I would have ended with this comment rather than starting with it.

File: packages/flutter/lib/src/material/slider.dart (hunk 2 above)
Comment (dleyba042) [reply packages/flutter/lib/src/material/slider.dart]: Noted, I like that, it is more concise.

File: packages/flutter/lib/src/material/slider.dart (hunk 4 above)
Comment (dleyba042) [reply packages/flutter/lib/src/material/slider.dart]: I think It goes thru a linter when you push to the main Flutter repo. I'll look at the style guide again.

File: packages/flutter/test/material/slider_test.dart (hunk 3 above)
Comment (dleyba042) [reply packages/flutter/test/material/slider_test.dart]: That makes sense, I will refactor.

File: packages/flutter/lib/src/material/slider.dart (hunk 1 above)
Comment (dleyba042) [reply packages/flutter/lib/src/material/slider.dart:688]: I asked on Discord and got this link and response: "This used to be a syntax error, but the requirement was thrown out when pattern matching was added (it behaves the same as if there was a break statement, unlike C-style switches)"
link-> https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#language-1

Comment (dwgray) [reply packages/flutter/lib/src/material/slider.dart:688]: Thanks Devin!  That makes total sense - and it looks like the 3.0 release is new enough that the rest of the documentation hasn't caught up.
